<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditDialog</class>
 <widget class="QDialog" name="EditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>653</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Recipe</string>
  </property>
  <property name="windowIcon">
   <iconset resource="anymeal.qrc">
    <normaloff>:/images/anymeal.png</normaloff>:/images/anymeal.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="editor_tab_widget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="general_tab">
      <attribute name="title">
       <string>&amp;General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="title_label">
           <property name="text">
            <string>&amp;Title:</string>
           </property>
           <property name="buddy">
            <cstring>title_edit</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="title_edit">
           <property name="maxLength">
            <number>60</number>
           </property>
           <property name="placeholderText">
            <string>title</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="categories_label">
           <property name="text">
            <string>&amp;Categories:</string>
           </property>
           <property name="buddy">
            <cstring>categories_edit</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="categories_edit">
           <property name="placeholderText">
            <string>category1, category2, ...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="servings_label">
           <property name="text">
            <string>&amp;Yield:</string>
           </property>
           <property name="buddy">
            <cstring>servings_spin</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QSpinBox" name="servings_spin">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>9999</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="servings_unit_edit">
             <property name="maxLength">
              <number>40</number>
             </property>
             <property name="placeholderText">
              <string>servings</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>164</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="ingredients_tab">
      <attribute name="title">
       <string>&amp;Ingredients</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9" stretch="1,0">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QTreeView" name="ingredients_view">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="uniformRowHeights">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPushButton" name="ingredient_group_button">
             <property name="text">
              <string>Add &amp;Group</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ingredient_button">
             <property name="text">
              <string>Add &amp;Ingredient</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="remove_ingredient_button">
             <property name="text">
              <string>&amp;Remove</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ingredient_up_button">
             <property name="text">
              <string>&amp;Up</string>
             </property>
             <property name="icon">
              <iconset resource="anymeal.qrc">
               <normaloff>:/images/up.svg</normaloff>:/images/up.svg</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="ingredient_down_button">
             <property name="text">
              <string>&amp;Down</string>
             </property>
             <property name="icon">
              <iconset resource="anymeal.qrc">
               <normaloff>:/images/down.svg</normaloff>:/images/down.svg</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QStackedWidget" name="ingredient_stack">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="ingredient_page">
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QGroupBox" name="ingredient_group">
             <property name="title">
              <string>I&amp;ngredient</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="amount_label">
                <property name="text">
                 <string>A&amp;mount:</string>
                </property>
                <property name="buddy">
                 <cstring>amount_type_combo</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QComboBox" name="amount_type_combo">
                  <item>
                   <property name="text">
                    <string>fraction</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>float</string>
                   </property>
                  </item>
                 </widget>
                </item>
                <item>
                 <widget class="QStackedWidget" name="amount_stack">
                  <property name="currentIndex">
                   <number>0</number>
                  </property>
                  <widget class="QWidget" name="fraction_page">
                   <layout class="QHBoxLayout" name="horizontalLayout_5">
                    <property name="leftMargin">
                     <number>0</number>
                    </property>
                    <property name="topMargin">
                     <number>0</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>0</number>
                    </property>
                    <item>
                     <widget class="QSpinBox" name="integer_spin">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                        <horstretch>1</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="maximum">
                       <number>99999</number>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="plus_label">
                      <property name="text">
                       <string>+</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="numerator_spin">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                        <horstretch>1</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="divide_label">
                      <property name="text">
                       <string>/</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="denominator_spin">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                        <horstretch>1</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="minimum">
                       <number>1</number>
                      </property>
                      <property name="maximum">
                       <number>100</number>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                  <widget class="QWidget" name="float_page">
                   <layout class="QHBoxLayout" name="horizontalLayout_4">
                    <property name="leftMargin">
                     <number>0</number>
                    </property>
                    <property name="topMargin">
                     <number>0</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>0</number>
                    </property>
                    <item>
                     <widget class="QDoubleSpinBox" name="amount_spin">
                      <property name="decimals">
                       <number>3</number>
                      </property>
                      <property name="maximum">
                       <double>99999.899999999994179</double>
                      </property>
                      <property name="singleStep">
                       <double>1.000000000000000</double>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="unit_label">
                <property name="text">
                 <string>Uni&amp;t:</string>
                </property>
                <property name="buddy">
                 <cstring>unit_combo</cstring>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QComboBox" name="unit_combo">
                <item>
                 <property name="text">
                  <string>per serving</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>small</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>medium</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>large</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>can</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>package</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>pinch</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>drop</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>dash</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>carton</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>bunch</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>slice</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>each</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>teaspoon</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>tablespoon</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>fluid ounce</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>cup</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>pint</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>quart</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>gallon</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ounce</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>pound</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>milliliter</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>cubic cm</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>centiliter</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>deciliter</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>liter</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>milligram</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>centigram</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>decigram</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>gram</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>kilogram</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string/>
                 </property>
                </item>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="name_label">
                <property name="text">
                 <string>Na&amp;me:</string>
                </property>
                <property name="buddy">
                 <cstring>name_edit</cstring>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="name_edit">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="placeholderText">
                 <string>ingredient</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="ingredient_section_page">
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QGroupBox" name="ingredient_section_group">
             <property name="title">
              <string>Ingredient &amp;Section</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QLabel" name="ingredient_section_label">
                  <property name="text">
                   <string>&amp;Title:</string>
                  </property>
                  <property name="buddy">
                   <cstring>ingredient_section_edit</cstring>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="ingredient_section_edit">
                  <property name="maxLength">
                   <number>60</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>67</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="instructions_tab">
      <attribute name="title">
       <string>In&amp;structions</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QTableView" name="instructions_view">
           <attribute name="horizontalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QPushButton" name="instructions_group_button">
             <property name="text">
              <string>&amp;Add Group</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="remove_instruction_button">
             <property name="text">
              <string>&amp;Remove</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="instructions_up_button">
             <property name="text">
              <string>&amp;Up</string>
             </property>
             <property name="icon">
              <iconset resource="anymeal.qrc">
               <normaloff>:/images/up.svg</normaloff>:/images/up.svg</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="instructions_down_button">
             <property name="text">
              <string>&amp;Down</string>
             </property>
             <property name="icon">
              <iconset resource="anymeal.qrc">
               <normaloff>:/images/down.svg</normaloff>:/images/down.svg</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="instruction_group">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Instruction</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QLabel" name="instruction_section_label">
              <property name="text">
               <string>&amp;Title</string>
              </property>
              <property name="buddy">
               <cstring>instruction_section_edit</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="instruction_section_edit">
              <property name="maxLength">
               <number>60</number>
              </property>
              <property name="placeholderText">
               <string>title</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="instructions_edit">
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="ok_button">
       <property name="text">
        <string>&amp;Ok</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancel_button">
       <property name="text">
        <string>&amp;Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>editor_tab_widget</tabstop>
  <tabstop>title_edit</tabstop>
  <tabstop>categories_edit</tabstop>
  <tabstop>servings_spin</tabstop>
  <tabstop>servings_unit_edit</tabstop>
  <tabstop>ingredients_view</tabstop>
  <tabstop>ingredient_group_button</tabstop>
  <tabstop>ingredient_button</tabstop>
  <tabstop>remove_ingredient_button</tabstop>
  <tabstop>ingredient_up_button</tabstop>
  <tabstop>ingredient_down_button</tabstop>
  <tabstop>amount_type_combo</tabstop>
  <tabstop>integer_spin</tabstop>
  <tabstop>numerator_spin</tabstop>
  <tabstop>denominator_spin</tabstop>
  <tabstop>amount_spin</tabstop>
  <tabstop>unit_combo</tabstop>
  <tabstop>name_edit</tabstop>
  <tabstop>ingredient_section_edit</tabstop>
  <tabstop>instructions_view</tabstop>
  <tabstop>instructions_group_button</tabstop>
  <tabstop>remove_instruction_button</tabstop>
  <tabstop>instructions_up_button</tabstop>
  <tabstop>instructions_down_button</tabstop>
  <tabstop>instruction_section_edit</tabstop>
  <tabstop>instructions_edit</tabstop>
 </tabstops>
 <resources>
  <include location="anymeal.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>amount_type_combo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>amount_stack</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ok_button</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>491</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>436</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancel_button</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>586</x>
     <y>520</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
